File: MatlabOctaveCodeSnippets/snip_transforms_DTFS_pulses.m
N=10; N1=5; k=0:N-1; %define durations and k
Xk=(1/N)*(sin(k*N1*pi/N)./sin(k*pi/N)) .* ...
    exp(-j*k*pi/N*(N1-1)); %obtain DTFS directly
Xk(1)=N1/N; %eliminate the NaN (not a number) in Xk(1)
%second alternative, via DFT. Generate x[n]:
xn=[ones(1,N1) zeros(1,N-N1)] %single period of x[n]
Xk2=fft(xn)/N %DTFS via DFT, Xk2 is equal to Xk

File: PythonCodeSnippets/snip_transforms_DTFS_pulses.py
import numpy as np
from numpy.fft import fft

# Define durations and k
N = 10
N1 = 5
k = np.arange(N)
            
# Obtain DTFS directly
Xk = (
    (1 / N)
    * (np.sin(k * N1 * np.pi / N) / np.sin(k * np.pi / N))
    * np.exp(-1j * k * np.pi / N * (N1 - 1))
)
Xk[0] = N1 / N  # Eliminate the NaN (not a number) in Xk[0]

# Second alternative, via DFT. Generate x[n]:
xn = np.concatenate((np.ones(N1), np.zeros(N - N1)))  # Single period of x[n]
Xk2 = fft(xn) / N  # DTFS via DFT, Xk2 is equal to Xk
